x-pack/test/functional/apps/rollup_job/rollup_jobs.js

```diff
@@ -4,64 +4,106 @@
+      for (const day of dates) {
+        await es.index({
+          index: `${prepend}-${day.format('MM-DD-YYYY')}`,
+          body: {
+            '@timestamp': day.toISOString(),
+            foo_metric: 1
+          }
+        });
+      }
+    }
 
-  describe('rollup job', function () {
-    this.tags('smoke');
     before(async () => {
-      // init data
-      await esArchiver.loadIfNeeded('logstash_functional');
-      await esArchiver.load('canvas/default');
+      await log.debug('make sure all dates have the same concept of "now"');
+      const now = new Date();
+      const pastDates = [
+        datemath.parse('now-1d', { forceNow: now }),
+        datemath.parse('now-2d', { forceNow: now }),
+        datemath.parse('now-3d', { forceNow: now }),
+      ];
+      await loadDates(pastDates);
       await PageObjects.common.navigateToApp('rollupJob');
-    });
 
-    after(async () => await esArchiver.unload('logstash_functional'));
+    });
 
-    it('create and save a new job', async () => {
-      const jobName = 'Testjob1';
-      const indexPattern = '.kibana*';
-      const indexName = 'rollup_index';
+    it('create new rollup job', async () => {
+      const indexPattern = 'to-be*';
       const interval = '1000ms';
 
-      await PageObjects.rollup.createNewRollUpJob();
-      await PageObjects.rollup.verifyStepIsActive(1);
-      await PageObjects.rollup.addRoleNameandIndexPattern(jobName, indexPattern);
-      await PageObjects.rollup.verifyIndexPatternAccepted();
-      await PageObjects.rollup.setIndexName(indexName);
-      await PageObjects.rollup.moveToNextStep();
-
-      //now navigate to histogram
-      await PageObjects.rollup.verifyStepIsActive(2);
-      await PageObjects.rollup.setJobInterval(interval);
-      await PageObjects.rollup.moveToNextStep();
-
-      //Terms (optional)
-      await PageObjects.rollup.verifyStepIsActive(3);
-      await PageObjects.rollup.moveToNextStep();
-
-      //Histogram(optional)
-      await PageObjects.rollup.verifyStepIsActive(4);
-      await PageObjects.rollup.moveToNextStep();
-
-      //Metrics(optional)
-      await PageObjects.rollup.verifyStepIsActive(5);
-      await PageObjects.rollup.moveToNextStep();
-
-      //saveJob and verify the name in the list
-      await PageObjects.rollup.verifyStepIsActive(6);
-      await PageObjects.rollup.saveJob();
-
-      // verify jobListTitle
-      const jobList = indexBy(await PageObjects.rollup.getJobList(), 'jobName');
-      log.debug(JSON.stringify(jobList));
-      log.debug(Object.keys(jobList));
-      expect(Object.keys(jobList)).to.have.length(1);
+      await PageObjects.rollup.createNewRollUpJob(rollupJobName, indexPattern, indexName,
+        interval, ' ', true, { time: '*/10 * * * * ?', cron: true });
+
+      await PageObjects.common.navigateToApp('indexManagement');
+
+      await log.debug('Wait a 10 seconds to ensure that the job has triggered.');
+      await PageObjects.common.sleep(10000);
```

Comment:
I would prefer pulling with state check over hardcoding 10 sec sleep as it may be an issue in future (e.g. any changes to functionality or network/CI worker slowness)
```
      await retry.waitForWithTimeout('Wait for job to be triggered', 10000,
         async () => {
           \\ get current job status and check it is completed?
           return state === 'completed';
         }
    );

Reply:
I was looking for this function. I can definitely do that.